File: java/lib/src/main/java/truelayer/signing/HeaderName.java
package truelayer.signing;

/**
 * Case-insensitive string header name comparison.
 */
class HeaderName {
    private final String name;

    HeaderName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        HeaderName that = (HeaderName) o;
        return name.equalsIgnoreCase(that.name);
    }

    @Override
    public int hashCode() {
        return name.toLowerCase().hashCode();
    }
}

File: kotlin/lib/src/main/kotlin/truelayer/signing/HeaderName.kt
package truelayer.signing

internal data class HeaderName(val name: String) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as HeaderName

        if (name.lowercase() != other.name.lowercase()) return false

        return true
    }

    override fun hashCode(): Int {
        return name.lowercase().hashCode()
    }
}